docs/changelog/4.2.rst

```diff
@@ -0,0 +1,1146 @@
+Changelogs for 4.2.x
+====================
+
+.. changelog::
+  :version: 4.2.0
+  :released: *unreleased*
+
+  This release *blah blah blah* …
+
+  Major features for this release:
+
+  * Lua records
+  * IXFRDist
+  * API changes
+
+  Minor features, weave paragraph from this:
+
+  * 6823: On incoming NOTIFY load our serial from backend to have it
+  * 6969: re-do of an earlier PR to add the glorious log-log histograms
+  * 6997: Name threads in the programs
+  * 6727: ALIAS: Respond SERVFAIL on non-NOERRORs from resolver
+  * 6910: support for OpenSSL 1.1.1's ed25519 and ed448
+  * 6975: Enhance query-logging with timing
+  * 6913: actually truncate truncated responses
+  * 6730: Implement a smarter dedup for filling packets in auth
+
+  Upgrade notes:
+
+  * autoserial removed
+  * command line util parameters (for scripts)
+  * BIND config files fixup on package upgrade
+  * GOST and Botan removed
+  * #6633?
+
+  Possible tags we can use below: Bug Fixes, Improvements, Internals, API, Packages, PostgreSQL, LDAP, MySQL, Tools, DNSSEC, Removed Features, New Features, BIND, DNSUpdate, IXFRDist.
+
+  .. change::
+    :tags: Improvements, Internals
+    :pullreq: 7245
+
+    Use a cache-able soa record for the serial check caused by a notify.
+
+  .. change::
+    :tags: Improvements
+    :pullreq: 6894
+
+    Improved Lua records - Added all selector, and backupSelector fallbacks
+
+  .. change::
+    :tags: Bug Fixes, IXFRDist
+    :pullreq: 7225
+
+    Preserve the correct TTL for SOA records
+
+  .. change::
+    :tags: Bug Fixes
+    :pullreq: 6823
+    :tickets: 6821
+
+    On incoming NOTIFY load our serial from backend to have it available during slave-check.
+
+    Also log ourserial to ease debugging.
+
+  .. change::
+    :tags: Improvements, API, DNSSEC
+    :pullreq: 5988
+
+    API: Add TSIG key manipulation endpoints
+
+  .. change::
+    :tags: Improvments
+    :pullreq: 7026
+
+    Configure --enable-pdns-option --with-third-party-module
```

Comment:
this is a breaking change - I've added it to the 4.2.0 important changes blogpost draft